# ChIP-Seq`_`analysis`_`pipeline
This repository documents ChIP-seq analysis pipeline for human and mouse data. 

## Input data
The input data (fastq files) resides in fq directory. The file name in fq
directory should be formatted as <{base`_`name}.fq.gz.

## Reference files
This script requires bowtie index and genome size files. Provide path to
these directories and files in the configuration file. 

## Configuration and cluster files
For human data, use hs prefix and for mouse data, use mm prefix configuration
files. Cluster file includes cluster-specific parameters. Do not edit this file
to run this pipeline.

## Wrapper script
Run the snakemake script dry and figure out the number of processes required in
the entire pipeline implementation. Edit -j flag for the number of jobs to be run
in wrapper script.

## Snake files
There are two snake files for a single pipeline. Part a includes preprocessing
the data (trim, fastQC, mapping, converting sam to bam, sorting, generating bed
files, and most importantly homer peak calling). The part b includes peak 
calling and generating visualization files using tools like macs1 and macs2. 

## JSON files
The part b script requires TREATMENT/CONTROL pairs. Those pairs are provided by
these two files.

## Running the script:
           dry run:
              < module load snakemake/6.8.2 >
              < snakemake -s chip-seq_mm10_a.snake >
           batch job:
              < sbatch chip-seq_snakemake_job.sh >

